# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

<div class="span5 alert alert-info">
### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
</div>
****

In [1]:
import pandas as pd
import numpy as np
from scipy import stats

In [2]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [3]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [4]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

## Notes and Observations
* There are 157 samples in the study, and so the CLT will apply
* Since the population standard deviation is not known, we will use the t-test to find
* If our null hypothesis, Canidates are not racially discriminated against by the spelling of their names, holds against 
* Our alternative hypothesis that black sounding names get fewer calls.

## Next steps
Perform statistical analysis to see if 'call' probability is higher for resumes with race 'w'

In [15]:
#Margin of error for sample
p = sum(data.call)/len(data.index)
n = len(data.index)
z = 1.96 #95% confidence level from Z table
std_error = np.sqrt(p*(1-p)/n)

margin_error = z*std_error

print("The margin of error for the sample is ", margin_error*100, "%")

The margin of error is  0.764095615545 %


In [23]:
#95% Confidence interval around the sample mean
s = np.std(data.call)
sigma = s/np.sqrt(n)
h0 = p #set the mean of our null hypothesis to be equal to p

lcl = h0 - (z*sigma)
ucl = h0 + (z*sigma)

lcl,ucl

(0.072851944446175973, 0.088133681837191571)

In [24]:
#P-value
h1 = sum(data[data.race=='b'].call)/len(data[data.race=='b'].index)

z = (h0 - h1)/sigma
z

4.1084591786942601

In [ ]:
#P = 0.00002

## Conclusions
Based on our analysis, if the null hypothesis was true, it would be __highly__ unlikely for the mean to be less than .072.
There is a .002% chance that the null hypothesis holds, and for this reason we can reject it.

That being said, further study is needed, such as ANOVA analysis, before it can be conclusively said that this 'race' variable
has a causal or correlated relationship to 'callback'